api/server/routes/messages.js

```diff
@@ -1,15 +1,21 @@
 const express = require('express');
 const router = express.Router();
 const { getMessages, likeMessage } = require('../../models/Message');
+const { increaseConvoViewCount, getSharedConvo } = require('../../models/Conversation');
 const requireJwtAuth = require('../../middleware/requireJwtAuth');
 
 router.get('/:conversationId', requireJwtAuth, async (req, res) => {
   const { conversationId } = req.params;
+  const convo = await getSharedConvo(conversationId);
+  if (convo?.user !== req.user.id) {
+    //increment view count only if the user is not the owner of the conversation
+    await increaseConvoViewCount(conversationId, req.user.id); //Added for https://github.com/aitok-ai/LibreChatInternal/issues/95
```

Comment:
I have found the source for all the repeated API calls. @Thomasbulls once implemented the like-a-message function. In his [implementation](https://github.com/aitok-ai/LibreChat/blob/3b8f043e48b4f3f603a0b6d48a4c018033110a47/client/src/components/Messages/Message.jsx#L128), he makes an API call for each message in a conversation to retrieve the liked status, therefore causing the view count in increment for each message in the conversation.

Reply:
that makes a lot of sense. I've found out that the number of GET /api/messages/ call scales with the number of messages inside the conversation.